src/utils/arrays.ts

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/**
+ * Determines if two arrays are different through a shallow comparison.
+ * @param a The first array. Must be defined.
+ * @param b The second array. Must be defined.
+ * @returns True if they are the same, false otherwise.
+ */
+export function arrayHasDiff(a: any[], b: any[]): boolean {
+    if (a.length === b.length) {
+        // When the lengths are equal, check to see if either array is missing
+        // an element from the other.
+        if (b.some(i => !a.includes(i))) return true;
```

Comment:
this looks `O(n^2)` when you could do it using sorting in `O(n log n)` or using a Set in amortized `O(n)`

Reply:
For the size of arrays we're passing through, the hit of a `Set` doesn't make sense to take on. We could reduce it to `O(n log n)` however, again, with the size of sets we're dealing with (tens of elements) the code complexity isn't worth the gain.

Calls to this are also guarded, and the function itself guards the potential performance hit as well. In rare scenarios would we ever actually hit this.

I'd be inclined to leave it as is and adjust it only if it shows up as a problem in performance profiles.

Reply:
Sane train of thought